snippets/SharedHandle.hpp

```diff
@@ -0,0 +1,404 @@
+#if !defined( VULKAN_HPP_NO_SMART_HANDLE )
+
+template <typename HandleType>
+class SharedHandleTraits;
+
+template <typename T, typename...>
+struct is_pool_free : std::false_type
+{
+};
+
+template <typename... Args>
+struct is_pool_free<PoolFreeShared<Args...>> : std::true_type
+{
+};
+
+template <typename T, typename... Args>
+VULKAN_HPP_CONSTEXPR VULKAN_HPP_INLINE bool is_pool_free_v = is_pool_free<T, Args...>::value;
+
+//=====================================================================================================================
+
+template <typename Header>
+class ControlBlockBase
+{
+public:
+  using header_type = Header;
```

Comment:
Removed. Those types are introduced because some sime I wanted to generalize on different type:
that one went from being
`using header_type = struct header<ParentType>`
but since I didn't want to rewrite the whole struct I just rewrite the `using`, and logic stays the same